File: arrays/container-with-most-water/container-with-most-water.java
class Solution {
  public int maxArea(int[] height) {
    int i = 0;
    int j = height.length - 1;
    int max = 0;

    while (i < j) {
      if (height[i] < height[j]) {
        max = Math.max((j - i) * height[i], max);
        i++;
      } else {
        max = Math.max((j - i) * height[j], max);
        j--;
      }
    }
    return max;
  }
}

File: arrays/container-with-most-water/container-with-most-water.go
package containerwithmostwater

func maxArea(height []int) int {
	i := 0
	j := len(height) - 1
	max := 0

	for i < j {
		if height[i] < height[j] {
			vol := (j - i) * height[i]

			if vol > max {
				max = vol
			}
			i++
		} else {
			vol := (j - i) * height[i]

			if vol > max {
				max = vol
			}
			j--
		}
	}
	return max
}
